File: extract_pdfs.js
import fs from 'fs';
import path from 'path';
import { getDocument } from 'pdfjs-dist/legacy/build/pdf.mjs';

const directoryPath = path.join(process.cwd(), 'src/kaynak/sari');

async function extractTextFromPDF(pdfPath) {
  try {
    const data = new Uint8Array(fs.readFileSync(pdfPath));
    const loadingTask = getDocument({ data });
    const pdf = await loadingTask.promise;
    
    const numPages = pdf.numPages;
    let text = '';
    // Sadece ilk 5 sayfayı okuyalım ki çok uzamasın
    const pagesToRead = Math.min(numPages, 5);
    
    for (let i = 1; i <= pagesToRead; i++) {
      const page = await pdf.getPage(i);
      const content = await page.getTextContent();
      const strings = content.items.map(item => item.str);
      text += `--- Page ${i} ---\n` + strings.join(' ') + '\n';
    }
    
    return text;
  } catch (error) {
    return `Error reading ${pdfPath}: ${error.message}`;
  }
}

async function main() {
  const files = fs.readdirSync(directoryPath).filter(f => f.endsWith('.pdf'));
  
  for (const file of files) {
    console.log(`\n=========================================`);
    console.log(`Analyzing: ${file}`);
    console.log(`=========================================`);
    const filePath = path.join(directoryPath, file);
    const text = await extractTextFromPDF(filePath);
    console.log(text.substring(0, 1500) + (text.length > 1500 ? '...\n[TRUNCATED]' : ''));
  }
}

main().catch(console.error);

File: extract_pdfs.py
import fitz  # PyMuPDF
import os

dir_path = "src/kaynak/sari"
for filename in os.listdir(dir_path):
    if filename.endswith(".pdf"):
        filepath = os.path.join(dir_path, filename)
        print(f"\n{'='*40}")
        print(f"File: {filename}")
        print(f"{'='*40}")
        try:
            doc = fitz.open(filepath)
            for i in range(min(5, doc.page_count)):
                page = doc[i]
                text = page.get_text()
                print(f"--- Page {i+1} ---")
                print(text[:300] + ("...\n" if len(text) > 300 else "\n"))
        except Exception as e:
            print(f"Error reading {filename}: {e}")
